src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -197,14 +217,17 @@ public bool TryGetSyntaxRoot(out SyntaxNode root)
         /// <summary>
         /// Gets the root node of the syntax tree asynchronously.
         /// </summary>
-        public async Task<SyntaxNode> GetSyntaxRootAsync(CancellationToken cancellationToken = default)
+        /// <returns>
+        /// The returned <see cref="SyntaxNode"/> will be null if <see cref="SupportsSyntaxTree"/> returns false.
```

Comment:
i'd prefer it throws in this case.  code that calls this won't be expecting the null either.

Reply:
So this is the shipped behavior, I'm just documenting it. If we're going to change it I want to do treat that as a separate change.

Reply:
And yes I realize this is absolutely a mistake in the API design but changing it may require a VS-wide test pass and also means code fixes that walk across solutions are broken, etc. It's not a trivial change.